plugin/pkg/scheduler/scheduler.go

```diff
@@ -89,6 +92,12 @@ func (s *Scheduler) Run() {
 
 func (s *Scheduler) scheduleOne() {
 	pod := s.config.NextPod()
+
+	// If the assumed pods store contains this pod there is a conflict,
+	// since the store can't contain a pod without a host, and NextPod
+	// only gives us pods without hosts.
+	s.config.Modeler.ForgetAssumedPod(pod)
```

Comment:
Ah, see the unittest. This is a problem when you delete and rescheduled the same pod. The problem is:
1. The old pod is removed from the scheduled pods queue on delete (via reflector)
2. The new pod is removed from the queued pods because of the pop (this is the rescheduled old pod, that we're trying to find a host for again).
3. The old pod (with the host assignment) remains in assumed pods, because we remove from assumed pods if the pod exists in either of the other 2 queues.

So if you only have one node, it'll fail a fitness predicate like host port because the machines->pods map contains the stale assumed pod.